projects/js-packages/publicize-components/src/hooks/use-schedule-post/index.tsx

```diff
@@ -0,0 +1,110 @@
+import { siteHasFeature } from '@automattic/jetpack-script-data';
+import { dispatch, useDispatch, useSelect } from '@wordpress/data';
+import { store as editorStore } from '@wordpress/editor';
+import { useCallback, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { store as socialStore } from '../../social-store';
+import { features } from '../../utils';
+import useSocialMediaMessage from '../use-social-media-message';
+
+type SchedulePostOptions = {
+	/** The connection IDs to share to. */
+	connectionIds: number[];
+	/** The message to share. */
+	message?: string;
+	/** The timestamp to schedule the share for. */
+	timestamp: number;
+};
+
+/**
+ * Hook to schedule a post for sharing to social media connections.
+ *
+ * @return {object} Object containing schedule functionality and state.
+ */
+export function useSchedulePost() {
+	const [ isScheduling, setIsScheduling ] = useState( false );
+
+	const { postId, isAutosaveablePost, isDirtyPost } = useSelect( select => {
+		const editorSelector = select( editorStore );
+
+		return {
+			postId: editorSelector.getCurrentPostId(),
+			isAutosaveablePost: editorSelector.isEditedPostAutosaveable(),
+			isDirtyPost: editorSelector.isEditedPostDirty(),
+		};
+	}, [] );
+
+	const { createScheduledShare } = useDispatch( socialStore );
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+	const { message } = useSocialMediaMessage();
+	const hasMediaFeatures =
+		siteHasFeature( features.IMAGE_GENERATOR ) || siteHasFeature( features.ENHANCED_PUBLISHING );
+
+	const savePost = dispatch( editorStore ).savePost;
+
+	const schedulePost = useCallback(
+		async ( { connectionIds, timestamp }: SchedulePostOptions ) => {
+			if ( ! connectionIds.length || ! timestamp ) {
+				return false;
+			}
+
+			setIsScheduling( true );
```

Comment:
I think there are some selectors like `isSavingEntityRecord` that we could use to determine the busy state if we wanted. https://github.com/WordPress/gutenberg/blob/trunk/packages/core-data/src/selectors.ts#L928

Reply:
be1e5f6